ns1/datadog_checks/ns1/check.py

```diff
@@ -0,0 +1,448 @@
+import errno
+import json
+import logging
+import os
+
+from requests.exceptions import ConnectionError, HTTPError, InvalidURL, Timeout
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+from .ns1_url_utils import Ns1Url
+
+
+class Ns1Check(AgentCheck):
+    NS1_SERVICE_CHECK = "ns1.can_connect"
+
+    def __init__(self, name, init_config, instances):
+        super(Ns1Check, self).__init__(name, init_config, instances)
+        self.api_endpoint = self.instance.get("api_endpoint")
+        if not self.api_endpoint:
+            raise ConfigurationError('NS1 API endpoint must be specified in configuration')
+
+        self.api_key = self.instance.get("api_key")
+        if not self.api_key:
+            raise ConfigurationError('NS1 API key must be specified in configuration')
+
+        self.headers = {"X-NSONE-Key": self.api_key}
+
+        self.metrics = self.instance.get("metrics")
+        if not self.metrics or len(self.metrics) == 0:
+            raise ConfigurationError('Invalid metrics config!')
+
+        self.query_params = self.instance.get("query_params")
+        self.usage_count_path = "/opt/datadog-agent/log"
+        self.usage_count_fname = 'ns1_usage_count.txt'
+        self.service_check(
+            self.NS1_SERVICE_CHECK,
+            AgentCheck.OK,
+            message='Configuration check for NS1 API endpoint %s was successful' % self.api_endpoint,
+        )
```

Comment:
That's not a right place for submitting the service check. This class is instantiated only once, so the `ok` service check will only be sent once. You want to move that inside the `check` method
```suggestion
```